src/analyses/goto_check.cpp

```diff
@@ -1324,17 +1327,21 @@ void goto_checkt::check_rec(const exprt &expr, guardt &guard, bool address)
   }
   else if(expr.id()==ID_and || expr.id()==ID_or)
   {
-    if(!expr.is_boolean())
+    if(expr.type().id() != ID_bool)
       throw "`"+expr.id_string()+"' must be Boolean, but got "+
             expr.pretty();
 
     guardt old_guard=guard;
 
     for(const auto &op : expr.operands())
     {
-      if(!op.is_boolean())
-        throw "`"+expr.id_string()+"' takes Boolean operands only, but got "+
-              op.pretty();
+      if(op.type().id() != ID_bool)
+      {
```

Comment:
This change seems not particularly useful; if you want to really have it, then please put into separate PR.